pkg/types/machinepools.go

```diff
@@ -21,6 +21,14 @@ const (
 	HyperthreadingDisabled HyperthreadingMode = "Disabled"
 )
 
+// Architecture is the instruction set architecture for the machines in a pool.
+type Architecture string
+
+const (
+	//ArchitectureAMD64 indicates AMD64 (x86_64)
```

Comment:
nit: missing a space after the `//` and [a trailing period][1].

[1]: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences